src/workerd/api/container.c++

```diff
@@ -14,7 +14,26 @@ namespace workerd::api {
 
 Container::Container(rpc::Container::Client rpcClient, bool running)
     : rpcClient(IoContext::current().addObject(kj::heap(kj::mv(rpcClient)))),
-      running(running) {}
+      running(running) {
+}
+
+void catchAndAbortOnDisconnect(jsg::Lock& js, jsg::Value err) {
+  jsg::JsValue handle = jsg::JsValue(err.getHandle(js));
+  auto kjException = js.exceptionToKj(handle);
+
+  // Check if the error is around disconnect
+  if (kjException.getType() == kj::Exception::Type::DISCONNECTED) {
+    auto e = JSG_KJ_EXCEPTION(FAILED, Error, "The client has disconnected");
+
+    // And abort the DO if it's due to a disconnect
+    IoContext::current().abort(kj::mv(e));
```

Comment:
Do we really want to abort? (honest question).
Shouldn't we just reconnect to CC?

Reply:
How would it look like reconnecting in EW?

I think aborting could make sense because it would be like a script update (disconnects are infrequent)